app/controllers/users/induction_coordinators/registrations_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class Users::InductionCoordinators::RegistrationsController < Devise::RegistrationsController
+  def start_registration; end
+
+  def confirm_school
+    @schools = School.where(id: params["school_ids"])
+    @email = params["email"]
+  end
+
+  def check_registration_email
+    @email = params["induction_coordinator_profile"]["email"]
+
+    @user = User.find_by(email: @email)
+
+    if @user
+      flash[:notice] = "You already have an account. Sign in"
+      redirect_to controller: "/users/sessions", action: :new, email: @email
+    else
+      domain = @email.split("@")[1]
+      @schools = School.where("'#{domain}' = ANY (domains)")
+
+      if @schools.any?
+        handle_matching_schools
+      else
+        redirect_to induction_coordinator_registration_check_email_path, alert: "No schools matched your email"
+      end
+    end
+  end
+
+  def handle_matching_schools
```

Comment:
👍 but I think this should go at the bottom under 'private' as it's just a helper rather than an action handler